HIP-XXXX.md

```diff
@@ -0,0 +1,298 @@
+|     RRSIG(example.www. DNSKEY)                                   |
++------------------------------------------------------------------+
++------------------------------------------------------------------+
+| Handshake CA: proofofconcept.                                    |
++------------------------------------------------------------------+       
+| Version: 3 (0x2)                                                 |
+| Serial Number: ............                                      |  
+| X509v3 extensions:                                               |
+|   X509v3 Name constraints:                                       |
+|     proofofconcept                                               |
+|   X509v3 Basic Constraints:                                      |
+|     CA:TRUE                                                      |
+|   Urkel: [                                                       |
+|     Root: 6c3078b26c3f7858d78bd2b63530d66b413f32f0ace083e8f21... |
+|     Proof: <proof>                                               |  
+|     Data: 0x............._delegation. NS                         | 
+|   ]                                                              |
++------------------------------------------------------------------+
+```
+
+
+The following is the internal TBS structure of an X509 certificate (RFC 5280).
+
+```
+TBSCertificate  ::=  SEQUENCE  {
+        version         [0]  EXPLICIT Version DEFAULT v1,
+        serialNumber         CertificateSerialNumber,
+        signature            AlgorithmIdentifier,
+        issuer               Name,
+        validity             Validity,
+        subject              Name,
+        subjectPublicKeyInfo SubjectPublicKeyInfo,
+        issuerUniqueID  [1]  IMPLICIT UniqueIdentifier OPTIONAL,
+                             -- If present, version MUST be v2 or v3
+        subjectUniqueID [2]  IMPLICIT UniqueIdentifier OPTIONAL,
+                             -- If present, version MUST be v2 or v3
+	extensions      [3]  EXPLICIT Extensions OPTIONAL
+                             -- If present, version MUST be v3		
+```
+
+In the extensions field, we can define custom extensions. Each extension has an OID (Object identifier, a unique id) called `extnID` and the value of the extension `extnValue` is an ASN.1 DER encoded structure. An ASN.1 OCTET STRING should be suitable for storing arbitrary data.
+
+
+
+### 1.0 Urkel proof extension:
+
+```
+extnId: TBD
+extnValue: n<urkel root><urkel proof>,...
+```
+
+The following sections describe what to include in the `extnValue` field depending on UPDATEs to the name within the last 32 urkel intervals.
+
+### 1.1 No UPDATEs:
+
+Certificates MUST be generated weekly and should include the 32nd most recent tree root and proof. This is the most common case.
+
+### 1.2 NS modified:
+
+Exactly the same as 1.1. Servers MUST NOT include the new NS records. If its a HIP-5 delegation, refer to 1.3.

```

Comment:
hmm the timing section says two proofs "_may be_ required" hopefully this is clear enough to mean not in all cases (but open to suggestions). Updates to plain NS records aren't useful to the client because they're not necessary to construct a validation path to the certificate. Advantage of not including those is sending only one proof instead of two and the old one is better since it accommodates out of sync clients.